bundles/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/internal/handler/SleepIQCloudHandler.java

```diff
@@ -262,6 +274,116 @@ public void setPauseMode(@Nullable String bedId, boolean mode) {
         }
     }
 
+    /**
+     * Get the foundation features of the specified bed
+     *
+     * @param bedId the bed identifier
+     */
+    public @Nullable FoundationFeaturesResponse getFoundationFeatures(@Nullable String bedId) {
+        if (bedId == null) {
+            return null;
+        }
+        try {
+            return cloud.getFoundationFeatures(bedId);
+        } catch (ResponseFormatException e) {
+            logger.debug("CloudHandler: Unable to parse foundation features response for bed={}", bedId);
+        } catch (SleepIQException e) {
+            logger.debug("CloudHandler: Exception getting foundation features, bed={}", bedId, e);
+        }
+        return null;
+    }
+
+    /**
+     * Get the foundation status of the specified bed
+     *
+     * @param bedId the bed identifier
+     */
+    public @Nullable FoundationStatusResponse getFoundationStatus(@Nullable String bedId) {
+        if (bedId == null) {
+            return null;
+        }
+        try {
+            return cloud.getFoundationStatus(bedId);
+        } catch (ResponseFormatException e) {
+            logger.debug("CloudHandler: Unable to parse foundation status response for bed={}", bedId);
+        } catch (SleepIQException e) {
+            logger.debug("CloudHandler: Exception getting foundation status, bed={}: {}", bedId, e.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * Set a foundation adjustment preset.
+     *
+     * @param bedId the bed identifier
+     * @param side the side of the bed
+     * @param preset the preset to be applied
+     * @param speed the speed with which to make the adjustment
+     */
+    public void setFoundationPreset(@Nullable String bedId, Side side, FoundationPreset preset,
+            FoundationActuatorSpeed speed) {
+        if (bedId == null) {
+            return;
+        }
+        try {
+            cloud.setFoundationPreset(bedId, side, preset, speed);
+        } catch (ResponseFormatException e) {
+            logger.debug("CloudHandler: ResponseFormatException setting foundation preset for bed={}: {}", bedId,
+                    e.getMessage());
+        } catch (SleepIQException e) {
+            logger.debug("CloudHandler: Exception setting the foundation preset for bed={}", bedId, e);
+        }
+        return;
+    }
+
+    /**
+     * Set a foundation position on head or foot of bed side.
+     *
+     * @param bedId the bed identifier
+     * @param side the side of the bed
+     * @param actuator the head or foot of the bed
+     * @param position the new position of the actuator
+     * @param speed the speed with which to make the adjustment
+     */
+    public void setFoundationPosition(@Nullable String bedId, Side side, FoundationActuator actuator, int position,
+            FoundationActuatorSpeed speed) {
+        if (bedId == null) {
```

Comment:
It doesn't make sense to call this method with `bedId == null`. Consider making it non-nullable and avoid calling it in this case.